net-vpn/i2p/i2p-0.9.32.ebuild

```diff
@@ -82,6 +76,20 @@ src_prepare() {
 	sed -i 's|clientApp.4.startOnLoad=true|clientApp.4.startOnLoad=false|' \
 		"${RES_DIR}/clients.config" || die
 
+	# generate classpath, keeping the default to be replaced later
+	local lib cp i=2
+	for lib in $(echo "$CP_DEPEND" | cut -d / -f 2 | sed 's|:0$||')
```

Comment:
This is a horrible idea. Now you can't modify `CP_DEPEND` at top (e.g. by adding min. version) without breaking this.

Reply:
The alternative is to have a patch for each version, depending on `CP_DEPEND`, but without compile time failure. The minimal version isn't needed for now, if it is ever needed, we can change it back to a patch, but IMO, for now, it's a not-so-bad fix and also really ease bumping.

Reply:
The `java-pkg_gen-cp` helper does this job in a very much better way. Simply call it with a variable name as an argument and it will assign the result to that variable.

Reply:
Hoo, nice, that would do the trick; hum, but it's tagged as internal, it would be unstable to use it.

Reply:
I can remove that tag, didn't notice that the original contributor had added it.

Reply:
Haha, my version was pretty hacky, I agree!
So, it's working (but not with a `local` variable; when it's no more internal, it would be good to document it). The main issue is that i2p use the wrapper to store many i2p specific config and defaults, it would be hard to phase out. Maybe we can provide this config directly and do not care about the classpath as it would already be generated during installation? I don't really know how it works, but we can try to remove it in the next bump.

Reply:
Minor nitpick, `for lib in ${i2p_cp//,/ }` is much nicer.

I'm not entirely sure what you're saying but I am aware that launching it via this wrapper is not ideal. We normally resolve the classpath at runtime, not build time, because jar filenames can potentially change. You're also referencing `/etc/java-config-2/current-system-vm`, which sucks a little because it doesn't respect the user VM, though I guess you're unlikely to set a user VM in this context.

Reply:
Done.

IMO, really removing the i2p wrapper would be good, but more fit the next bump, as there is already a bunch of changes now. I'll try to come up with something working in a few months (or if sooner, a revbump) if that works for you.